cadence/contracts/PrizeSavings.cdc

```diff
@@ -3535,13 +3634,17 @@ access(all) contract PrizeSavings {
             let endedRoundID = self.activeRound.getRoundID()
             let roundDuration = self.activeRound.getDuration()
             
+            // Set the actual end time on the ending round
+            // This is the moment we're finalizing - used for TWAB calculations
+            self.activeRound.setActualEndTime(now)
+            
             // Create new round starting now (before the swap)
             // Gap interactors are handled by lazy fallback - no explicit initialization needed
             let newRoundID = endedRoundID + 1
             let newRound <- create Round(
                 roundID: newRoundID,
                 startTime: now,
-                duration: roundDuration
+                configuredDuration: roundDuration
```

Comment:
shouldn't it use the self.config.drawIntervalSeconds here? So, if it's updated it will be reflected on the next round